fhd_core/HEALPix/healpix_cnv_generate.pro

```diff
@@ -90,27 +100,29 @@ ENDIF ELSE BEGIN
 ENDELSE
 
 ; Test for pixels past the horizon. We don't need to be precise with this, so turn off precession, etc..
-Eq2Hor,pix_ra, pix_dec, obs.JD0, alt_arr, az_arr, nutate=0,precess=0,aberration=0, refract=0, lon=obs.lon, alt=obs.alt, lat=obs.lat
-horizon_i = where(alt_arr LE 0, n_horizon, complement=h_use)
-IF n_horizon GT 0 THEN BEGIN
-    print,String(format='("Cutting ",A, " HEALPix pixels that were below the horizon.")',Strn(n_horizon))
-    xv_hpx = xv_hpx[h_use]
-    yv_hpx = yv_hpx[h_use]
-    hpx_inds = hpx_inds[h_use]
-ENDIF
+;Eq2Hor,pix_ra, pix_dec, obs.JD0, alt_arr, az_arr, nutate=0,precess=0,aberration=0, refract=0, lon=obs.lon, alt=obs.alt, lat=obs.lat
+;horizon_i = where(alt_arr LE 0, n_horizon, complement=h_use)
+;IF n_horizon GT 0 THEN BEGIN
+;    print, "im  here"
+;    print,String(format='("Cutting ",A, " HEALPix pixels that were below the horizon.")',Strn(n_horizon))
+;    xv_hpx = xv_hpx[h_use]
+;   yv_hpx = yv_hpx[h_use]
+;    hpx_inds = hpx_inds[h_use]
+;ENDIF
 
 x_frac=1.-(xv_hpx-Floor(xv_hpx))
 y_frac=1.-(yv_hpx-Floor(yv_hpx))
 
 min_bin=Min(Floor(xv_hpx)+dimension*Floor(yv_hpx))>0L
-max_bin=Max(Ceil(xv_hpx)+dimension*Ceil(yv_hpx))<(dimension*elements-1L)
+max_bin=Max(Ceil(xv_hpx)+dimension*Ceil(yv_hpx))<(dimension*elements-1L) 
+; give hist some data, min and max for bins, binsize=1?
 h00=histogram(Floor(xv_hpx)+dimension*Floor(yv_hpx),min=min_bin,max=max_bin,/binsize,reverse_ind=ri00)
 h01=histogram(Floor(xv_hpx)+dimension*Ceil(yv_hpx),min=min_bin,max=max_bin,/binsize,reverse_ind=ri01)
 h10=histogram(Ceil(xv_hpx)+dimension*Floor(yv_hpx),min=min_bin,max=max_bin,/binsize,reverse_ind=ri10)
 h11=histogram(Ceil(xv_hpx)+dimension*Ceil(yv_hpx),min=min_bin,max=max_bin,/binsize,reverse_ind=ri11)
+
```

Comment:
The above code is calculating the four corners surrounding every pixel, and putting the results in difficult-to-understand lookup tables (the "reverse indices").